mindcv/models/pit.py

```diff
@@ -0,0 +1,516 @@
+            kernel_size=patch_size,
+            stride=stride,
+            pad_mode="pad",
+            padding=padding,
+            has_bias=True,
+        )
+
+    def construct(self, x: Tensor) -> Tensor:
+        x = self.conv(x)
+        return x
+
+
+class conv_head_pooling(nn.Cell):
+    """define pooling layer using conv in spatial tokens with an additional fully-connected layer
+    (to adjust the channel size to match the spatial tokens)"""
+
+    def __init__(self,
+                 in_feature: int,
+                 out_feature: int,
+                 stride: int,
+                 pad_mode: str = "pad") -> None:
+        super().__init__()
+        self.conv = nn.Conv2d(
+            in_feature,
+            out_feature,
+            kernel_size=stride + 1,
+            padding=stride // 2,
+            stride=stride,
+            pad_mode=pad_mode,
+            group=in_feature,
+            has_bias=True,
+        )
+        self.fc = nn.Dense(in_channels=in_feature, out_channels=out_feature, has_bias=True)
+
+    def construct(self, x, cls_token):
+        x = self.conv(x)
+        cls_token = self.fc(cls_token)
+
+        return x, cls_token
+
+
+class Attention(nn.Cell):
+    """define multi-head self attention block"""
+
+    def __init__(self,
+                 dim: int,
+                 num_heads: int = 8,
+                 qkv_bias: bool = False,
+                 attn_drop: float = 0.0,
+                 proj_drop: float = 0.0) -> None:
+        super().__init__()
+        self.num_heads = num_heads
+        head_dim = dim // num_heads
+        self.scale = head_dim**-0.5
+        # get pair-wise relative position index for each token inside the window
+        self.q = nn.Dense(in_channels=dim, out_channels=dim, has_bias=qkv_bias)
+        self.k = nn.Dense(in_channels=dim, out_channels=dim, has_bias=qkv_bias)
+        self.v = nn.Dense(in_channels=dim, out_channels=dim, has_bias=qkv_bias)
+        self.attn_drop = nn.Dropout(keep_prob=1 - attn_drop)
+        self.proj = nn.Dense(dim, dim)
+        self.proj_drop = nn.Dropout(keep_prob=1 - proj_drop)
+        self.softmax = nn.Softmax(axis=-1)
+
+        self.batchmatmul = ops.BatchMatMul()
+
+    def construct(self, x):
+        B, N, C = x.shape
+        q = ops.reshape(self.q(x), (B, N, self.num_heads, C // self.num_heads)) * self.scale
+        q = ops.transpose(q, (0, 2, 1, 3))
+        k = ops.reshape(self.k(x), (B, N, self.num_heads, C // self.num_heads))
+        k = ops.transpose(k, (0, 2, 3, 1))
+        v = ops.reshape(self.v(x), (B, N, self.num_heads, C // self.num_heads))
+        v = ops.transpose(v, (0, 2, 1, 3))
+
+        attn = self.batchmatmul(q, k)
+        attn = self.softmax(attn)
+        attn = self.attn_drop(attn)
+
+        x = self.batchmatmul(attn, v)
+        x = ops.reshape(ops.transpose(x, (0, 2, 1, 3)), (B, N, C))
+        x = self.proj(x)
+        x = self.proj_drop(x)
+        return x
+
+
+class DropPath(nn.Cell):
```

Comment:
why not use droppath in layers

Reply:
we use droppath1D to specify the dim of input for the readability.

Reply:
We didn't notice it before.Already changed